mpitests/test_apply.py

```diff
@@ -468,3 +543,95 @@ def circuit_mpi():
 
         local_state_vector = circuit_mpi()
         assert np.allclose(local_state_vector, local_expected_output_cpu, atol=tol, rtol=0)
+
+
+class TestExpval:
```

Comment:
Do we also have tests for multiple observables in the expval? Like `qml.expval(qml.PauliX(0) @ qml.PauliY(n-1))`. We should be sure to check the tensor observables work here too. Similarly, Hamiltonians need to be tested somewhere.

Reply:
Done!